cardano-node/src/Cardano/Node/Run.hs

```diff
@@ -205,6 +231,158 @@ handlePeersListSimple tr nodeKern = forever $ do
+  nt <- either (\err -> panic $ "Cardano.Node.Run.handleSimpleNodeNonP2P.readTopologyFile: " <> err) pure eitherTopology
+
+  let diffusionTracers :: DiffusionTracers
+      diffusionTracers = createDiffusionTracers nodeTracers
+
+      (ipProducerAddrs, dnsProducerAddrs) = producerAddressesNonP2P nt
+
+      dnsProducers :: [DnsSubscriptionTarget]
+      dnsProducers = [ DnsSubscriptionTarget (Text.encodeUtf8 addr) port v
+                     | (NodeAddress (NodeHostDnsAddress addr) port, v) <- dnsProducerAddrs
+                     ]
+
+      ipProducers :: IPSubscriptionTarget
+      ipProducers = IPSubscriptionTarget
+                     [ toSockAddr (addr, port)
+                     | (NodeAddress (NodeHostIPAddress addr) port) <- ipProducerAddrs
+                     ]
+                     0
+      diffusionArguments :: DiffusionArguments m
+      diffusionArguments =
+        createDiffusionArgumentsNonP2P
+          publicIPv4SocketOrAddr
+          publicIPv6SocketOrAddr
+          localSocketOrPath
+          (ncDiffusionMode nc)
+          ipProducers
+          dnsProducers
+
+  ipv4 <- traverse getSocketOrSocketInfoAddr publicIPv4SocketOrAddr
+  ipv6 <- traverse getSocketOrSocketInfoAddr publicIPv6SocketOrAddr
+
+  traceNamedObject
+    (appendName "addresses" trace)
+    (meta, LogMessage . Text.pack . show $ catMaybes [ipv4, ipv6])
+  traceNamedObject
+    (appendName "diffusion-mode" trace)
+    (meta, LogMessage . Text.pack . show . ncDiffusionMode $ nc)
+  traceNamedObject
+    (appendName "dns-producers" trace)
+    (meta, LogMessage . Text.pack . show $ dnsProducers)
+  traceNamedObject
+    (appendName "ip-producers" trace)
+    (meta, LogMessage . Text.pack . show $ ipProducers)
+
+  withShutdownHandling nc trace $ \sfds ->
+    void $
+      Node.run
+        DisabledP2PMode
+        RunNodeArgs
+          { rnTraceConsensus = consensusTracers nodeTracers
+          , rnTraceNTN       = nodeToNodeTracers nodeTracers
+          , rnTraceNTC       = nodeToClientTracers nodeTracers
+          , rnProtocolInfo   = pInfo
+          , rnNodeKernelHook = \registry nodeKernel -> do
+              maybeSpawnOnSlotSyncedShutdownHandler nc sfds trace registry
+                (Node.getChainDB nodeKernel)
+              onKernel nodeKernel
+          }
+        StdRunNodeArgs
+          { srnBfcMaxConcurrencyBulkSync   = unMaxConcurrencyBulkSync <$> ncMaxConcurrencyBulkSync nc
+          , srnBfcMaxConcurrencyDeadline   = unMaxConcurrencyDeadline <$> ncMaxConcurrencyDeadline nc
+          , srnChainDbValidateOverride     = ncValidateDB nc
+          , srnSnapshotInterval            = ncSnapshotInterval nc
+          , srnDatabasePath                = dbPath
+          , srnDiffusionArguments          = diffusionArguments
+          , srnDiffusionTracers            = diffusionTracers
+          , srnEnableInDevelopmentVersions = False -- ncTestEnableDevelopmentNetworkProtocols nc
```

Comment:
What's the plan here?

Reply:
Well spotted! I forgot to add that config option! On it 😄

Reply:
@nfrisby what about now?